mpas_analysis/ocean/variable_stream_map.py

```diff
@@ -0,0 +1,46 @@
+# mappings of stream names from various MPAS-O versions to those in
+# mpas_analysis
+oceanStreamMap = {'timeSeriesStats': ['timeSeriesStatsOutput',
+                                      'timeSeriesStatsMonthly',
+                                      'timeSeriesStatsMonthlyOutput']}
+
+
+oceanVariableMap = {}
```

Comment:
Ah, so each entry in the dictionary is a list, and, yes, the order of the list is such that earlier entries are preferred over later entries.  This doesn't require an `OrderedDict` because it's the list entries, not the dictionary keys, that need to be ordered by preference.  We don't try anything to do with the dictionary keys when we're searching for matching variables, we go through the list in order.

Also, you're worrying about efficiency for a part of the code with negligible impact on runtime, I guarantee you.